docs/testing.md

```diff
@@ -0,0 +1,146 @@
+Every scenario that does not work in OCIS, is tagged with `@skipOnOcis` and additionally should be marked with an issue number e.g. `@issue-ocis-20`.
+This tag means that this particular scenario is skipped because of [issue no 20 in the ocis repository](https://github.com/owncloud/ocis/issues/20).
+Additionally, some issues have scenarios that demonstrate the current buggy behaviour in ocis(reva) and are skipped on oC10.
+Have a look into the [documentation](https://doc.owncloud.com/server/developer_manual/testing/acceptance-tests.html#writing-scenarios-for-bugs) to understand why we are writing those tests.
+
+If you want to work on a specific issue
+
+1.  run the tests marked with that issue tag
+
+    E.g.:
+    ```
+    make test-acceptance-api \
+    TEST_SERVER_URL=http://localhost:9140 \
+    TEST_EXTERNAL_USER_BACKENDS=true \
+    TEST_OCIS=true \
+    OCIS_REVA_DATA_ROOT=/var/tmp/reva/ \
+    BEHAT_FILTER_TAGS='~@skipOnOcV10&&~@skipOnLDAP&&@TestAlsoOnExternalUserBackend&&~@local_storage&&@issue-ocis-20'
+    ```
+
+    Note that the `~@skipOnOcis` tag is replaced by `~@skipOnOcV10` and the issue tag `@issue-ocis-20` is added.
+    We want to run all tests that are skipped in CI because of this particular bug, but we don't want to run the tests
+    that demonstrate the current buggy behaviour.
+
+2.  the tests will fail, try to understand how and why they are failing
+3.  fix the code
+4.  go back to 1. and repeat till the tests are passing.
+5.  adjust tests that demonstrate the **buggy** behaviour
+
+    delete the tests in core that are tagged with that particular issue and `@skipOnOcV10`, but be careful because a lot of tests are tagged with multiple issues.
+    Only delete tests that demonstrate the buggy behaviour if you fixed all bugs related to that test. If not you might have to adjust the test.
+6.  unskip tests that demonstrate the **correct** behaviour
+
+    The `@skipOnOcis` tag should not be needed now, so delete it, but leave the issue tag for future reference.
+7.  make a PR to core with the changed tests
+8.  make a PR to ocis running the adjusted tests
+
+    To confirm that all tests (old and changed) run fine make a PR to ocis with your code changes and point drone to your branch in core to get the changed tests.
+    For that change this line in the `acceptance-tests` section
+
+    `'git clone -b master --depth=1 https://github.com/owncloud/core.git /srv/app/testrunner',`
+
+    to clone your core branch e.g.
+
+    `'git clone -b fixRevaIssue122 --depth=1 https://github.com/owncloud/core.git /srv/app/testrunner',`
+
+9.  merge PRs
+
+    After you have confirmed that the tests pass everywhere merge the core PR and immediately revert the change in 8. and merge the ocis PR
+
+    If the changes also affect the `ocis-reva` repository make sure the changes get ported over there immediately, otherwise the tests will start failing there.
```

Comment:
IMO it it the other way around. Ocis is just a "runtime". The changes are always made in ocis-reva first

Reply:
what about previews? they are in ocis but not in ocis-reva
since a week or so we have preview tests, they would not affect reva

Reply:
yes. so we have different scenarios